features-support/step_definitions/features.js

```diff
@@ -41,30 +28,18 @@ function getScenarioFromProject(callback, world) {
       return;
     }
 
-    // Get a link to the last feature on the page.
-    try {
-      var featureLinkRegex = /class="spec-link" href="([\w\/.?=-]+\.feature[\w\/.?=-]+)"/g;
-      while((featureLink = featureLinkRegex.exec(world.body)) !== null) {
-        featureLinks.push(featureLink[1]);
-      }
-      featureLink = featureLinks[featureLinks.length - 1];
-    } catch(error) {
-      callback(error);
-      return;
-    }
-
-    var featureUrl = 'http://localhost:' + world.appPort + featureLink;
+    world.browser.findElements(By.css('.spec-link'))
+    .then(function (specLinks) {
+      var featureUrl = 'http://localhost:' + world.appPort + featureLink;
 
-    // Follow the link.
-    request.get(featureUrl, function(error, response, body) {
-      if (error) {
-        callback(error);
-        return;
-      }
+      var featureLink = specLinks[specLinks.length - 1];
 
-      world.statusCode = response.statusCode;
```

Comment:
Some errors, for instance in routing logic, will manifest as 404s, so it's probably useful to get the response code if we can... I do seem to recall reading that the webdriver protocol didn't support that (because it was supposed to be about UI, not networks) but they recognised there was customer demand for it so maybe it's possible now...